tests/chebyshev_test.py

```diff
@@ -0,0 +1,823 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from functools import partial
+import numpy as np
+from numpy.polynomial import chebyshev as ncs
+import numpy.polynomial.polyutils as npu
+from numpy.polynomial.polyutils import RankWarning
+
+from absl.testing import absltest
+from absl.testing import parameterized
+
+from jax import numpy as jnp
+from jax._src import test_util as jtu
+import jax.numpy.polynomials as pu
+from jax.numpy.polynomials import chebyshev as cs
+
+from jax.config import config
+config.parse_flags_with_absl()
+
+all_dtypes = jtu.dtypes.floating + jtu.dtypes.integer + jtu.dtypes.complex
+
+class TestChebyshev(jtu.JaxTestCase):
+
+  @parameterized.named_parameters(jtu.cases_from_list(
+    {"testcase_name": "_dtype_off={}_dtype_scl={}_mul{}".format(
+      dtype_off, dtype_scl, mul),
+    "dtype_off": dtype_off, "dtype_scl": dtype_scl, "mul": mul}
+    for dtype_off in all_dtypes
+    for dtype_scl in all_dtypes
+    for mul in [0,1]))
+  def testChebline(self, dtype_off, dtype_scl, mul):
+    rng = jtu.rand_default(self.rng())
+
+    def args_maker():
+      off = rng(1, dtype_off)
+      scl = rng(1, dtype_scl)
+      scl  = (scl*mul).astype(dtype_scl)
+      return (off, scl)
+
+    jnp_fn = partial(cs.chebline, trim=False)
+
+    with jtu.strict_promotion_if_dtypes_match([dtype_off, dtype_scl]):
+      self._CheckAgainstNumpy(ncs.chebline, cs.chebline, args_maker, check_dtypes=False,
+                              tol=1e-6)
```

Comment:
One reason I'm having trouble landing this, I think, is because you've hard-coded tolerances like `1E-6` throughout. When the tolerance is unspecified, it defaults to an appropriate tolerance for a given dtype. When you override this with a single number, it uses that number regardless of the dtype. I think tests may behave better if you remove these default tolerances; or if you must specify them, do something like `tol={np.float32: 1e-6}` to change the default tolerance only for a particular dtype.

Reply:
Alright, I left tolerances unspecified for all tests except for a few. For those few, I specified tolerances by dtype as done in lax-numpy-test.